### PR TITLE
Move provider to EC

### PR DESCRIPTION
**Pull request by MoonNight131**
Replaced GC/GCore/etc to EC/Edgecenter/etc
Updated docs
Updated examples
Updated dependecies
Move Go version to 1.19

### CODE REVIEW

**Comment by alex2304** [comment]
Please change `gcorelabs.com` to `edgecenter.ru` in Description of resources and in docs
image

File: docs/index.md [hunk 1]
```diff
@@ -17,14 +17,16 @@ terraform {
   required_version = ">= 0.13.0"
   required_providers {
     edgecenter = {
-      source  = "Edge-Center/edgecenter"
-      version = "0.3.18"
+      source = "local.edgecenter.ru/repo/edgecenter"
```
Comment (alex2304) [inline docs/index.md:20]: What is the source address? Is this a path to a folder, or an url?

File: docs/index.md [hunk 2]
```diff
@@ -17,14 +17,16 @@ terraform {
...
+      version = "0.1.0"
     }
   }
 }
 
 provider edgecenter {
   permanent_api_token = "251$d3361.............1b35f26d8"
+  edgecenter_platform_api = "https://api.edgecenter.ru/iam"
```
Comment (alex2304) [inline docs/index.md:28]: Why do you need to set these variables explicitly? Seems that it will be set by default in `provider.go`

**Review by alex2304 — CHANGES_REQUESTED** [review]
Seems OK for me, please finish the changes.

And special thanks for splitting commits!

File: docs/index.md (hunk 2 above)
Comment (MoonNight131) [reply docs/index.md:28]: It's for docs only

File: docs/index.md (hunk 1 above)
Comment (MoonNight131) [reply docs/index.md:20]: Terraform registry or path to folder in .terraform.d/plugins

File: docs/index.md (hunk 2 above)
Comment (alex2304) [reply docs/index.md:28]: Ok, just an example, got it

**Review by alex2304 — APPROVED** [review]

**Pull request merged**